src/System.Device.Gpio/System/Device/Gpio/GpioDriver.cs

```diff
@@ -82,6 +82,12 @@ protected internal virtual void SetPinMode(int pinNumber, PinMode mode, PinValue
     /// <returns>The value of the pin.</returns>
     protected internal abstract PinValue Read(int pinNumber);
 
+    /// <summary>
+    /// Toggle the current value of a pin.
+    /// </summary>
+    /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+    protected internal abstract void Toggle(int pinNumber);
```

Comment:
That's a real breaking change (binary as well as source). I think this should have a default implementation instead.

Reply:
Why not `PinValue Toggle(int pinNumber)`?
I don't know if the hw implementation provides the value, but in the other cases it is certainly available.
If the hardware driver does not provide the value, we can decide to either return null or read the value and return it. 
I prefer this signature to avoid future breaking changes in case the hw driver will provide the value in the next releases.

Reply:
> That's a real breaking change (binary as well as source). I think this should have a default implementation instead.

Some hardware provide this feature, so does not. We agreeded that the braking change is ok as it then surfaced on the GpioController following the API call. And the best way to handle the change is to move it to the driver because it's the **only** place tracking really the state of the pins.